src/renderer/helpers/utils.js

```diff
@@ -672,3 +672,35 @@ export function escapeHTML(untrusted) {
 export function deepCopy(obj) {
   return JSON.parse(JSON.stringify(obj))
 }
+
+/**
+ * Check if the `name` of the error is `TimeoutError` to know if the error was caused by a timeout or something else.
+ * @param {number} milliseconds
+ * @param {RequestInfo|URL} input
+ * @param {RequestInit=} init
+ */
+export async function fetchWithTimeout(milliseconds, input, init) {
+  const timeoutSignal = AbortSignal.timeout(milliseconds)
```

Comment:
Not blocking issue but good to name parameters as their function (what to store)
For small functions this is fine (so it's not blocking) but it's a good practice for bigger functions
```suggestion
export async function fetchWithTimeout(timeoutMs, input, init) {
  const timeoutSignal = AbortSignal.timeout(timeoutMs)
```

Reply:
@absidue want to implement this?